### PR TITLE
Add custom exception class and helper classes for promptflow-evals

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/microsoft/promptflow/blob/main/CONTRIBUTING.md).**
- [ ] **I confirm that all new dependencies are compatible with the MIT license.**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/_exceptions.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+"""This includes enums and classes for exceptions for use in promptflow-evals."""
+
+from enum import Enum
+
+from azure.core.exceptions import AzureError
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/_exceptions.py:8]: Is azure core already a dependency of pf ?

File: src/promptflow-evals/promptflow/evals/_exceptions.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    """
+
+    def __init__(
+        self,
+        message: str,
+        no_personal_data_message: str,
+        *args,
+        target: ErrorTarget = ErrorTarget.UNKNOWN,
+        category: ErrorCategory = ErrorCategory.UNKNOWN,
+        blame: ErrorBlame = ErrorBlame.UNKNOWN,
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/_exceptions.py:78]: Would customer see this property when exception is raised?

File: src/promptflow-evals/promptflow/evals/_exceptions.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+class ErrorTarget(Enum):
+    """Error target to be specified when using PromptflowEvalsException class.
+
+    When using PromptflowEvalsException, specify the code are that was being targeted when the
+    exception was triggered.
+    """
+
+    UNKNOWN = "Unknown"
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/_exceptions.py:52]: Would we continue to add `ErrorTarget` as we find them ?

File: src/promptflow-evals/promptflow/evals/_exceptions.py (hunk 1 above)
Comment (diondrapeck) [reply src/promptflow-evals/promptflow/evals/_exceptions.py:8]: Yep: https://github.com/microsoft/promptflow/blob/f0b01d502fb89d615436be27f9bb1c62753dd402/src/promptflow-evals/pyproject.toml#L53

File: src/promptflow-evals/promptflow/evals/_exceptions.py (hunk 3 above)
Comment (diondrapeck) [reply src/promptflow-evals/promptflow/evals/_exceptions.py:52]: Yes, exactly. I will add as I start going through the refactoring.

File: src/promptflow-evals/promptflow/evals/_exceptions.py (hunk 2 above)
Comment (diondrapeck) [reply src/promptflow-evals/promptflow/evals/_exceptions.py:78]: Nope - all they see is the "message" attribute:

image

File: src/promptflow-evals/promptflow/evals/_exceptions.py (hunk 1 above)
Comment (singankit) [reply src/promptflow-evals/promptflow/evals/_exceptions.py:8]: I will check on this.

**Review by singankit — DISMISSED** [review]

**Comment by w-javed** [comment]
LGTM

**Review by w-javed — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/_exceptions.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    :param error_category: The error category, defaults to Unknown.
+    :type error_category: ~promptflow.evals._exceptionsErrorCategory
+    :param error: The original exception if any.
+    :type error: Exception
+    """
+
+    def __init__(
+        self,
+        message: str,
+        no_personal_data_message: str,
```
[the remark this answers is not in the document]
Comment (diondrapeck) [reply src/promptflow-evals/promptflow/evals/_exceptions.py]: done!

**Review by singankit — APPROVED** [review]

**Review by w-javed — APPROVED** [review]

**Review by MilesHolland — APPROVED** [review]

**Pull request merged**